packages/cli/src/util/promote/status.ts

```diff
@@ -0,0 +1,265 @@
+import chalk from 'chalk';
+import type {
+  LastAliasRequest,
+  Deployment,
+  PaginationOptions,
+  Project,
+} from '@vercel-internals/types';
+import type Client from '../client';
+import elapsed from '../output/elapsed';
+import formatDate from '../format-date';
+import getDeployment from '../get-deployment';
+import { getPkgName } from '../pkg-name';
+import getScope from '../get-scope';
+import ms from 'ms';
+import sleep from '../sleep';
+import getProjectByNameOrId from '../projects/get-project-by-id-or-name';
+import { ProjectNotFound } from '../errors-ts';
+import renderAliasStatus from '../alias/render-alias-status';
+
+interface DeploymentAlias {
+  alias: {
+    alias: string;
+    deploymentId: string;
+  };
+  id: string;
+  status: 'completed' | 'in-progress' | 'pending' | 'failed';
+}
+
+interface AliasesResponse {
+  aliases: DeploymentAlias[];
+  pagination: PaginationOptions;
+}
+
+/**
+ * Continuously checks a deployment status until it has succeeded, failed, or
+ * taken longer than the timeout (default 3 minutes).
+ *
+ * @param {Client} client - The Vercel client instance
+ * @param {string} [contextName] - The scope name; if not specified, it will be
+ * extracted from the `client`
+ * @param {Deployment} [deployment] - Info about the deployment which is used
+ * to display different output following a promotion request
+ * @param {Project} project - Project info instance
+ * @param {string} [timeout] - Milliseconds to poll for succeeded/failed state
```

Comment:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (soft-blocking):** clarify value

In this or a follow-up PR: Can we say something more specific than "Milliseconds" here? That's not strictly true. It can be any format that `ms` accepts.
‏
```suggestion
 * @param {string} [timeout] - Milliseconds (or `ms` shorthand) to poll for succeeded/failed state
```